cli/src/Api.ts

```diff
@@ -111,4 +142,110 @@ export default class Api {
             .signAndSend(account);
         return txHash;
     }
+
+    // Working groups
+    // TODO: This is a lot of repeated logic from "/pioneer/joy-roles/src/transport.substrate.ts"
+    // (although simplified a little bit)
+    // Hopefully this will be refactored to "joystream-js" soon
+    protected singleLinkageResult<T extends Codec>(result: LinkageResult) {
+        return result[0] as T;
+    }
+
+    protected multiLinkageResult<K extends Codec, V extends Codec>(result: LinkageResult): [Vec<K>, Vec<V>] {
+        return [ result[0] as Vec<K>, result[1] as Vec<V> ];
+    }
+
+    protected workingGroupApiQuery(group: WorkingGroups) {
+        const module = apiModuleByGroup[group];
+        return this._api.query[module];
+    }
+
+    protected async memberProfileById(memberId: MemberId, throwError = true): Promise<Profile | null> {
+        const profile = await this._api.query.members.memberProfile(memberId) as Option<Profile>;
+
+        if (throwError && profile.isNone) {
```

Comment:
Why is the function throwing if a member id doesn't exist. Shouldn't the caller just test if the return value is null?
In general I think it makes more sense for the function to throw if we add a check for the existence of a key in a map (by reading directly from storage with api.rpc.getStorage)

Reply:
I think you're right, I copied some code from Pioneer's `joy-roles` transport here and probably got too incluenced by the fact that a similar method was throwing an error there.

It seems to be not too good design though, because then we have to use TypeScript assertions to "convince" the compiler that the value is never `null` when `throwError === true`.

Fixed in a new commit.